File: solutions/XYZCoder/solve/cpp/XYZCoderSolver.cpp
#include <fstream>
#include <topcoder>
#include "XYZCoder.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    int room; tc::read(ifs, room); tc::next(ifs);
    int size; tc::read(ifs, size);
    ifs.close();

    std::ofstream ofs(argv[2]);
    XYZCoder solver;
    tc::show(ofs, solver.countWays(room, size));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}

File: solutions/XYZCoder/solve/ruby/XYZCoderSolver.rb
#! /usr/bin/env ruby

require_relative "XYZCoder"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  room = reader.next(TInt)
  reader.next()
  size = reader.next(TInt)

  result = XYZCoder.new().countWays(room, size)
  IO.write(ARGV[1], Writer.new.next(result, TInt).to_s)
end

init
main
